test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -846,5 +848,59 @@ public void TensorFlowTransformCifarInvalidShape()
         }
+
+        /// <summary>
+        /// Class to hold features and predictions.
+        /// </summary>
+        public class TensorFlowSentiment
+        {
+            public string Sentiment_Text;
+            [VectorType(600)]
+            public int[] Features;
+            [VectorType(2)]
+            public float[] Prediction;
+        }
+
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))]
+        public void TensorFlowSentimentClassificationTest()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+            var data = new[] { new TensorFlowSentiment() { Sentiment_Text = "this film was just brilliant casting location scenery story direction everyone's really suited the part they played and you could just imagine being there robert  is an amazing actor and now the same being director  father came from the same scottish island as myself so i loved the fact there was a real connection with this film the witty remarks throughout the film were great it was just brilliant so much that i bought the film as soon as it was released for  and would recommend it to everyone to watch and the fly fishing was amazing really cried at the end it was so sad and you know what they say if you cry at a film it must have been good and this definitely was also  to the two little boy's that played the  of norman and paul they were just brilliant children are often left out of the  list i think because the stars that play them all grown up are such a big profile for the whole film but these children are amazing and should be praised for what they have done don't you think the whole story was so lovely because it was true and was someone's life after all that was shared with us all" } };
+            var dataView = mlContext.Data.ReadFromEnumerable(data);
+
+            var lookupMap = mlContext.Data.ReadFromTextFile(@"sentiment_model\imdb_word_index.csv",
+                   columns: new[]
+                   {
+                        new TextLoader.Column("Words", DataKind.TX, 0),
+                        new TextLoader.Column("Ids", DataKind.I4, 1),
+                   },
+                   separatorChar: ','
+               );
+
+            // We cannot resize variable length vector to fixed lenght vector in ML.Net
+            // The trick here is to create two pipelines.
+            // The first pipeline tokenzies the strings into words and maps the words to an integer which is an index in the dictionary.
+            // Then this integer vector is retrieved from the pipeline and resized to fixed length.
+            // The second pipeline takes the resized integer vector and passed to TensoFlow and get the classification scores.
+            var estimator = mlContext.Transforms.Text.TokenizeWords("Sentiment_Text", "TokenizedWords")
+                .Append(mlContext.Transforms.Conversion.ValueMap(lookupMap, "Words", "Ids", new[] { ("TokenizedWords", "Features") }));
```

Comment:
>ValueMap [](start = 56, length = 8)

- Is this transform doing the re-sizing to the fixed length ?

- Does it matter what the fixed length is ?   I presume the model was built with  fixed  length  shaped input.  But I do not see the shape specified

Reply:
resizing is done at 893 in C# code.

---
In reply to: [251976925](https://github.com/dotnet/machinelearning/pull/2302#discussion_r251976925) [](ancestors = 251976925)

Reply:
I do not see shape specified in line 893

---
In reply to: [251977773](https://github.com/dotnet/machinelearning/pull/2302#discussion_r251977773) [](ancestors = 251977773,251976925)

Reply:
sorry not 893. Its line 899

---
In reply to: [251978523](https://github.com/dotnet/machinelearning/pull/2302#discussion_r251978523) [](ancestors = 251978523,251977773,251976925)

Reply:
We are calling the `Predict` API  and hence in Line #899  we can resize  the output of the first pipeline.

How would this work for the 'Transform' API... where the testData  has 2 rows (say)

row1 ->  "Hi"  ->  dimension 50 (say)
row2 -> "<some long sentence>  ->  dimension 5000 (say)

Will it work ?

---
In reply to: [251978928](https://github.com/dotnet/machinelearning/pull/2302#discussion_r251978928) [](ancestors = 251978928,251978523,251977773,251976925)

Reply:
I understand we are not training the TF model 

I wanted to know if the re-size  operation on dataview would be supported  --  If it is supported, can we add it to the unit test ?

This test case  does single prediction (use of `Predict` API)  but does not show use  of  the `Transform` API  where  we would have to re-size  more than just 1  row of  variable length vector.  

---
In reply to: [252003040](https://github.com/dotnet/machinelearning/pull/2302#discussion_r252003040) [](ancestors = 252003040,251981308,251978928,251978523,251977773,251976925)